libpdf/models/horizontal_box.py

```diff
@@ -0,0 +1,173 @@
+"""Definition of HorizontalBox to contain text in the PDF."""
+
+from typing import List
+
+
+class Char:  # pylint: disable=too-few-public-methods # simplicity is good.
+    """
+    Define the character class.
+
+    :ivar ~.text: a plain char of the chararcter
+    :vartype text: str
+    :ivar x0: distance from the left of the page to the left edge of the character
+    :vartype x0: float
+    :ivar y0: distance from the bottom of the page to the lower edge of the character (less than y1)
+    :vartype y0: float
+    :ivar x1: distance from the left of the page to the right edge of the character
+    :vartype x1: float
+    :ivar y1: distance from the bottom of the page to the upper edge of the character (greater than y0)
+    :vartype y1: float
+    """
+
+    def __init__(
+        self,
+        text: str,
+        x0: float = None,
+        y0: float = None,
+        x1: float = None,
+        y1: float = None,
+    ):
+        """Init the class with plain char of a character and its rectangular coordinates."""
+        self.x0 = x0
+        self.y0 = y0
+        self.x1 = x1
+        self.y1 = y1
+        self.text = text
+
+    def __repr__(self):
+        """Make the text part of the repr for better debugging."""
+        return f'{type(self).__name__}({self.text})'
+
+
+class Word:
+    """
+    Define the word class.
+
+    A word shall contain several characters.
+
+    :ivar chars: a list of the chararcter
+    :vartype chars: List[Char]
+    """
+
+    def __init__(
+        self,
+        chars: List[Char],
+        x0: float = None,
+        y0: float = None,
+        x1: float = None,
+        y1: float = None,
+    ):
+        """Init the class with plain text of a word and its rectangular coordinates."""
+        self.x0 = x0
+        self.y0 = y0
+        self.x1 = x1
+        self.y1 = y1
+        self.chars = chars
+        if self.chars:
+            self.get_coordinates()
+
+    @property
+    def text(self):
+        """Return plain text of a list of chararcters."""
+        return ''.join([x.text for x in self.chars])
+
+    def get_coordinates(self):
```

Comment:
Should be a private function as self.x/y/0/1 is directly exposed. Also why make it a function if it is only used once?